Rename RenderSelectionInfo to RenderSelectionGeometry as part of rename work to clarify classes around highlights.

megangardner opened the pull request:
#### 5b2b95668aa27f1659143dd2509ce2495bc42cd1
<pre>
Rename RenderSelectionInfo to RenderSelectionGeometry as part of rename work to clarify classes around highlights.
<a href="https://bugs.webkit.org/show_bug.cgi?id=261625">https://bugs.webkit.org/show_bug.cgi?id=261625</a>
rdar://115575135

Reviewed by Simon Fraser.

Several classes associated with Highlights have a very confusing naming
structure. This is the second rename of several to help clarify
what these classes are and their relationship to each other.

* Source/WebCore/rendering/RenderSelectionGeometry.cpp: Renamed from Source/WebCore/rendering/RenderSelectionInfo.cpp.
(WebCore::RenderSelectionGeometryBase::RenderSelectionGeometryBase):
(WebCore::RenderSelectionGeometryBase::repaintRectangle):
(WebCore::RenderSelectionGeometry::RenderSelectionGeometry):
(WebCore::RenderSelectionGeometry::repaint):
(WebCore::RenderBlockSelectionGeometry::RenderBlockSelectionGeometry):
(WebCore::RenderBlockSelectionGeometry::repaint):
* Source/WebCore/rendering/RenderSelectionGeometry.h: Renamed from Source/WebCore/rendering/RenderSelectionInfo.h.
(WebCore::RenderSelectionGeometryBase::repaintContainer const):
(WebCore::RenderSelectionGeometryBase::state const):
(WebCore::RenderSelectionGeometry::collectedSelectionQuads const):
(WebCore::RenderSelectionGeometry::rect const):
(WebCore::RenderBlockSelectionGeometry::rects const):

Canonical link: <a href="https://commits.webkit.org/268051@main">https://commits.webkit.org/268051@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fcc6a8e792fff99affffce15bb43589b6d91c6e4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18523 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18857 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19462 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20379 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17325 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18718 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22170 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19002 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19203 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18747 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18922 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16133 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21255 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16144 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16895 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23348 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17163 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17065 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21243 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17642 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/14950 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16702 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4402 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21066 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17484 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/rendering/SelectionRangeData.cpp:
```diff
@@ -176,16 +176,16 @@ IntRect SelectionRangeData::collectBounds(ClipToVisibleContent clipToVisibleCont
             while (block && !is<RenderView>(*block)) {
                 LOG_WITH_STREAM(Scrolling, stream << " added block " << *block);
-                std::unique_ptr<RenderSelectionInfo>& blockInfo = renderers.add(block, nullptr).iterator->value;
+                std::unique_ptr<RenderSelectionGeometry>& blockInfo = renderers.add(block, nullptr).iterator->value;
```
`auto` would have saved you the rename.

smfr commented on Source/WebCore/rendering/SelectionRangeData.cpp:
```diff
@@ -255,7 +255,7 @@ void SelectionRangeData::apply(const RenderRange& newSelection, RepaintMode bloc
     for (auto* currentRenderer = selectionStart; currentRenderer && currentRenderer != selectionEnd; currentRenderer = selectionIterator.next()) {
         if (isValidRendererForSelection(*currentRenderer, m_renderRange)) {
-            std::unique_ptr<RenderSelectionInfo> selectionInfo = makeUnique<RenderSelectionInfo>(*currentRenderer, true);
+            std::unique_ptr<RenderSelectionGeometry> selectionInfo = makeUnique<RenderSelectionGeometry>(*currentRenderer, true);
```
`auto`, and call it `selectionGeometry`

smfr commented on Source/WebCore/rendering/SelectionRangeData.cpp:
```diff
@@ -265,10 +265,10 @@ void SelectionRangeData::apply(const RenderRange& newSelection, RepaintMode bloc
             auto* containingBlock = currentRenderer->containingBlock();
             while (containingBlock && !is<RenderView>(*containingBlock)) {
-                std::unique_ptr<RenderBlockSelectionInfo>& blockInfo = newSelectedBlocks.add(containingBlock, nullptr).iterator->value;
+                std::unique_ptr<RenderBlockSelectionGeometry>& blockInfo = newSelectedBlocks.add(containingBlock, nullptr).iterator->value;
```
`auto& blockSelectionGeometry`

megangardner committed:
Rename RenderSelectionInfo to RenderSelectionGeometry as part of rename work to clarify classes around highlights.
https://bugs.webkit.org/show_bug.cgi?id=261625
rdar://115575135

Reviewed by Simon Fraser.

Several classes associated with Highlights have a very confusing naming
structure. This is the second rename of several to help clarify
what these classes are and their relationship to each other.

* Source/WebCore/rendering/RenderSelectionGeometry.cpp: Renamed from Source/WebCore/rendering/RenderSelectionInfo.cpp.
(WebCore::RenderSelectionGeometryBase::RenderSelectionGeometryBase):
(WebCore::RenderSelectionGeometryBase::repaintRectangle):
(WebCore::RenderSelectionGeometry::RenderSelectionGeometry):
(WebCore::RenderSelectionGeometry::repaint):
(WebCore::RenderBlockSelectionGeometry::RenderBlockSelectionGeometry):
(WebCore::RenderBlockSelectionGeometry::repaint):
* Source/WebCore/rendering/RenderSelectionGeometry.h: Renamed from Source/WebCore/rendering/RenderSelectionInfo.h.
(WebCore::RenderSelectionGeometryBase::repaintContainer const):
(WebCore::RenderSelectionGeometryBase::state const):
(WebCore::RenderSelectionGeometry::collectedSelectionQuads const):
(WebCore::RenderSelectionGeometry::rect const):
(WebCore::RenderBlockSelectionGeometry::rects const):

Canonical link: https://commits.webkit.org/268051@main
